### PR TITLE
Add support of CC and BCC into SendEmailMessage Helper

### PR DESCRIPTION
**Pull request by alexandcote**
Support CC and BCC into SendEmailMessage Helper

### CODE REVIEW

**Comment by c-schmitz** [comment]
Same here: please undo the .gitignore changes.

File: .gitignore [hunk 1]
```diff
@@ -4,6 +4,9 @@
 # ignore CAssetManager for debugging
 /framework/web/CAssetManager.php
 
+# vendor directory
```
Comment (alexandcote) [inline .gitignore]: I add this because if you use composer to manage your plugins, you will have a `vendor` folder to ignore.

File: .gitignore [hunk 2]
```diff
@@ -79,12 +82,6 @@ Thumbs.db
 .buildpath
 *.un~
 launch.json
-# ignore plugins directory by default except LS plugin
```
Comment (alexandcote) [inline .gitignore]: This was a duplicated entry in the `.gitignore` file. You will find the same code [here](https://github.com/LimeSurvey/LimeSurvey/pull/945/files#diff-a084b794bc0759e7a6b77810e01874f2L29)

File: .gitignore (hunk 1 above)
Comment (TonisOrmisson) [reply .gitignore]: vendor dir is third-party
https://github.com/LimeSurvey/LimeSurvey/blob/a1cd6d20c7a5bd22b90cb8c4c3680fa432e18fc5/composer.json#L34

Comment (alexandcote) [reply .gitignore]: I delete this entry in the `.gitignore` file.

File: composer.json [hunk 3]
```diff
@@ -3,14 +3,15 @@
 	"description" : "The most popular FOSS online survey tool on the web",
 	"homepage" : "https://www.limesurvey.org/",
 	"license" : "GPL 2.0",
+	"type": "limesurvey",
```
Comment (TonisOrmisson) [inline composer.json]: https://getcomposer.org/doc/04-schema.md#type

**Comment by Shnoulle** [comment]
Think we must create a new SendMail Class and add an event in this one. Global SendMail event for ALL event.

The hard point are : what information this event need (more than to/from/subject/body … 

1. Mail was sent to 'admin'/'particpant'/'token'
2. This mail was 'type' email (password for admin, invite for token etc …

See [eventBeforeTokenEmailExtended](https://framagit.org/SondagePro-LimeSurvey-plugin/eventBeforeTokenEmailExtended) this one already add CC and BCC to Token email, but not for admin and other email.

File: composer.json (hunk 3 above)
Comment (c-schmitz) [reply composer.json]: ?

Comment (alexandcote) [reply composer.json]: It was a mistake... sorry about that

**Comment by c-schmitz** [comment]
This needs to go to the develop branch.

**Pull request closed without merging**